demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,552 @@
+
+/**
+ * @brief Definition of the HTTP network context.
+ */
+struct HTTPNetworkContext
+{
+    int tcpSocket;
+};
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Performs a DNS lookup on the given host name then establishes a TCP
+ * connection to the server.
+ *
+ * @param[in] pServer Host name of server.
+ * @param[in] port Server port.
+ * @param[out] pTcpSocket Pointer to TCP socket file descriptor.
+ *
+ * @return EXIT_FAILURE on failure; EXIT_SUCCESS on success.
+ */
+static int connectToServer( const char * pServer,
+                            uint16_t port,
+                            int * pTcpSocket );
+
+/**
+ * @brief The transport send function that defines the transport interface.
+ *
+ * This is passed to the #HTTPTransportInterface.send function and used to
+ * send data over the network.
+ *
+ * @param[in] pContext User defined context (TCP socket for this demo).
+ * @param[in] pBuffer Buffer containing the bytes to send over the network stack.
+ * @param[in] bytesToSend Number of bytes to write to the network.
+ *
+ * @return Number of bytes sent; negative value on error.
+ */
+static int32_t transportSend( HTTPNetworkContext_t * pContext,
+                              const void * pBuffer,
+                              size_t bytesToSend );
+
+/**
+ * @brief The transport receive function that defines the transport interface.
+ *
+ * @param[in] pContext User defined context (TCP socket for this demo).
+ * @param[out] pBuffer Buffer to read network data into.
+ * @param[in] bytesToRead Number of bytes requested from the network.
+ *
+ * This is passed to the #HTTPTransportInterface.recv function and used to
+ * receive data over the network.
+ *
+ * @return Number of bytes received; negative value on error.
+ */
+static int32_t transportRecv( HTTPNetworkContext_t * pContext,
+                              void * pBuffer,
+                              size_t bytesToRecv );
+
+/**
+ * @brief Send an HTTP request based on a specified method and path.
+ *
+ * @param[in] pTransportInterface The transport interface for making network calls.
+ * @param[in] pHost The host name of the server.
+ * @param[in] pMethod The HTTP request method.
+ * @param[in] pPath The Request-URI to the objects of interest.
+ *
+ * @return EXIT_FAILURE on failure; EXIT_SUCCESS on success.
+ */
+static int sendHttpRequest( HTTPTransportInterface_t * pTransportInterface,
+                            const char * pHost,
+                            const char * pMethod,
+                            const char * pPath );
+
+/*-----------------------------------------------------------*/
+
+static int connectToServer( const char * pServer,
+                            uint16_t port,
+                            int * pTcpSocket )
+{
+    int returnStatus = EXIT_SUCCESS;
+    struct addrinfo hints, * pIndex, * pListHead = NULL;
+    struct sockaddr * pServerInfo;
+    uint16_t netPort = htons( port );
```

Comment:
Does this raise a warning for C89 compliance? C89 does not allow having declarations and "code" together (unfortunately)

Reply:
I think it should not. It is just initializing variable, but good point. We should make sure compiler runs with c89 option. Out of scope for this PR though.

Reply:
Does not raise a warning. I am using the following C flags:
```
-Wall -Wextra -Wstrict-prototypes -Wold-style-definition -Wmissing-prototypes -Wmissing-declarations -Wdeclaration-after-statement
```

Reply:
The declarations and code together only pop up when you put "non-assignment code" like asserts, before the declarations, it does not apply to assignment type function usages.